clients/src/main/java/org/apache/kafka/common/network/SslTransportLayer.java

```diff
@@ -461,7 +463,7 @@ private void handshakeFinished() throws IOException {
             if (netWriteBuffer.hasRemaining())
                 key.interestOps(key.interestOps() | SelectionKey.OP_WRITE);
             else {
-                state = sslEngine.getSession().getProtocol().equals("TLSv1.3") ? State.POST_HANDSHAKE : State.READY;
+                state = sslEngine.getSession().getProtocol().equals(TLS13) ? State.POST_HANDSHAKE : State.READY;
                 key.interestOps(key.interestOps() & ~SelectionKey.OP_WRITE);
                 SSLSession session = sslEngine.getSession();
```

Comment:
This should move above the TLS check and be reused there (on the off-chance the getSession() call is more expensive than it looks).

Reply:
(In case it's not clear from the code snippet:  new line 468 should move up above the earlier addition, then reuse `session` in that line.)